systemtest/src/test/java/io/strimzi/systemtest/LogSettingST.java

```diff
@@ -479,6 +482,138 @@ void testJSONFormatLogging() {
+                        "\n" +
+                        "rootLogger.level = TRACE\n" +
+                        "rootLogger.appenderRefs = stdout\n" +
+                        "rootLogger.appenderRef.console.ref = STDOUT\n" +
+                        "rootLogger.additivity = false"))
+                .build();
+
+        ConfigMap configMapUo = new ConfigMapBuilder()
+                .withNewMetadata()
+                .withName("external-configmap-uo")
+                .withNamespace(NAMESPACE)
+                .endMetadata()
+                .addToData(Collections.singletonMap("log4j2.properties", "name = UOConfig\n" +
+                        "\n" +
+                        "appender.console.type = Console\n" +
+                        "appender.console.name = STDOUT\n" +
+                        "appender.console.layout.type = PatternLayout\n" +
+                        "appender.console.layout.pattern = [%d] %-5p <%-12.12c{1}:%L> [%-12.12t] %m%n\n" +
+                        "\n" +
+                        "rootLogger.level = TRACE\n" +
+                        "rootLogger.appenderRefs = stdout\n" +
+                        "rootLogger.appenderRef.console.ref = STDOUT\n" +
+                        "rootLogger.additivity = false"))
+                .build();
+
+        kubeClient().getClient().configMaps().inNamespace(NAMESPACE).createOrReplace(configMapTo);
+        kubeClient().getClient().configMaps().inNamespace(NAMESPACE).createOrReplace(configMapUo);
+
+        ExternalLogging elTo = new ExternalLoggingBuilder()
+                .withName("external-configmap-to")
+                .build();
+
+        ExternalLogging elUo = new ExternalLoggingBuilder()
+                .withName("external-configmap-uo")
+                .build();
+
+        // change to external logging
+        KafkaResource.replaceKafkaResource(CLUSTER_NAME, k -> {
+            k.getSpec().getEntityOperator().getTopicOperator().setLogging(elTo);
+            k.getSpec().getEntityOperator().getUserOperator().setLogging(elUo);
+        });
+
+        String finalEoPodName1 = eoPodName;
+        TestUtils.waitFor("Logger change", Constants.GLOBAL_POLL_INTERVAL, Constants.GLOBAL_TIMEOUT, () -> {
+                return cmdKubeClient().execInPodContainer(finalEoPodName1, "topic-operator", "cat", "/opt/topic-operator/custom-config/log4j2.properties").out().contains("rootLogger.level = TRACE")
+                        && cmdKubeClient().execInPodContainer(finalEoPodName1, "user-operator", "cat", "/opt/user-operator/custom-config/log4j2.properties").out().contains("rootLogger.level = TRACE");
+            }
+        );
+
+        eoPodName = eoPods.keySet().iterator().next();
+        eoPodHash = eoPods.get(eoPodName);
+        eoPods = DeploymentUtils.depSnapshot(eoName);
+        eoPodNameAfterLoggingChange = eoPods.keySet().iterator().next();
+        assertThat("Pod name changed after logging changed", eoPodNameAfterLoggingChange.equals(eoPodName), is(true));
```

Comment:
This is f*cking confusing as hell. Is it testing that pod name did or didn't changed? I guess it is that it didn't?

Reply:
Sorry about that. Yes, it is testing the name is not changed. As EO is deployment, RU=new suffix for the pod name. So if the RU did not happen (it should not when dynamically changing), the name should remain.